pkg/make/builder.go

```diff
@@ -467,6 +468,50 @@ func (b *Builder) BuildCmd(src, output string) func(context.Context) error {
 	}
 }
 
+// getVersion uses `git describe` to determine the version of the
+// running code. The underlying functionality is as simple as
+// strings.TrimPrefix, but there is some additional sanity checking
+// with a regex.
+func (b *Builder) getVersion(ctx context.Context) (string, error) {
+	gitVersion, err := b.execOut(ctx, "git", "describe", "--tags", "--always", "--dirty")
+	if err != nil {
+		return "", errors.Wrap(err, "git describe")
+	}
+
+	// The `-` is included in the "additional" part of the regex,
+	// to make the later concatenation correct. If, and when, we
+	// move to a windows style 0.0.0.0 format, this will need to
+	// change.
+	versionRegex, err := regexp.Compile(`^v?(\d+)\.(\d+)(?:\.(\d+))(?:(-.+))?$`)
+	if err != nil {
+		return "", errors.Wrap(err, "bad regex")
+	}
+
+	// regex match and check the results
+	matches := versionRegex.FindAllStringSubmatch(gitVersion, -1)
+
+	if len(matches) == 0 {
+		return "", errors.Errorf(`Version "%s" did not match expected format`, gitVersion)
```

Comment:
appending something like `: <major>.<minor>.<patch>.<additional>` to the error message might be helpful

Reply:
Okay